pkg/controller/v1beta1/inferenceservice/reconcilers/deployment/deployment_reconciler.go

```diff
@@ -0,0 +1,199 @@
+	componentStatus v1beta1.ComponentStatusSpec
+}
+
+func NewDeploymentReconciler(client client.Client,
+	scheme *runtime.Scheme,
+	componentMeta metav1.ObjectMeta,
+	componentExt *v1beta1.ComponentExtensionSpec,
+	podSpec *corev1.PodSpec,
+	componentStatus v1beta1.ComponentStatusSpec) *DeploymentReconciler {
+	return &DeploymentReconciler{
+		client:          client,
+		scheme:          scheme,
+		Deployment:      createRawDeployment(componentMeta, componentExt, podSpec, componentStatus),
+		componentExt:    componentExt,
+		componentStatus: componentStatus,
+	}
+}
+
+func createRawDeployment(componentMeta metav1.ObjectMeta,
+	componentExt *v1beta1.ComponentExtensionSpec,
+	podSpec *corev1.PodSpec,
+	componentStatus v1beta1.ComponentStatusSpec) *appsv1.Deployment {
+	var minReplicas int32
+	if componentExt.MinReplicas == nil {
+		minReplicas = int32(constants.DefaultMinReplicas)
+	} else {
+		minReplicas = int32(*componentExt.MinReplicas)
+	}
+
+	if minReplicas < int32(constants.DefaultMinReplicas) {
+		minReplicas = int32(constants.DefaultMinReplicas)
+	}
+	setDefaultPodSpec(podSpec)
+	deployment := &appsv1.Deployment{
+		ObjectMeta: componentMeta,
+		Spec: appsv1.DeploymentSpec{
+			Replicas: &minReplicas,
+			Selector: &metav1.LabelSelector{
+				MatchLabels: map[string]string{
+					"app": constants.GetRawServiceLabel(componentMeta.Name, componentMeta.Namespace),
+				},
+			},
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: map[string]string{
+						"app": constants.GetRawServiceLabel(componentMeta.Name, componentMeta.Namespace),
+					},
+				},
+				Spec: corev1.PodSpec(*podSpec),
+			},
+		},
+	}
+	setDefaultDeploymentSpec(&deployment.Spec)
+	return deployment
+}
+
+//checkNativeDeploymentExist checks is the deployment is existed?
+func (r *DeploymentReconciler) checkDeploymentExist(client client.Client) (constants.CheckResultType, error) {
+	//get deployment
+	existingDeployment := &appsv1.Deployment{}
+	err := client.Get(context.TODO(), types.NamespacedName{
+		Namespace: r.Deployment.Namespace,
+		Name:      r.Deployment.Name,
+	}, existingDeployment)
+	if err != nil {
+		if apierr.IsNotFound(err) {
+			return constants.CheckResultCreate, nil
+		}
+		return constants.CheckResultUnknown, err
+	}
+	//existed, check equalment
+	if semanticDeploymentEquals(r.Deployment, existingDeployment) {
+		return constants.CheckResultExisted, nil
+	}
+	return constants.CheckResultUpdate, nil
+}
+
+func semanticDeploymentEquals(desired, existing *appsv1.Deployment) bool {
+	return equality.Semantic.DeepEqual(desired.Spec, existing.Spec)
+}
+
+func setDefaultPodSpec(podSpec *corev1.PodSpec) {
+	if podSpec.DNSPolicy == "" {
+		podSpec.DNSPolicy = corev1.DNSClusterFirst
+	}
+	if podSpec.RestartPolicy == "" {
+		podSpec.RestartPolicy = corev1.RestartPolicyAlways
+	}
+	if podSpec.TerminationGracePeriodSeconds == nil {
+		TerminationGracePeriodSeconds := int64(corev1.DefaultTerminationGracePeriodSeconds)
```

Comment:
Knative defaults this to timeout value as you want to wait for all the in flight requests to finish before terminating the pod.